### DISCUSSION TITLE
[Feature] Improved face interactions in photos

### DISCUSSION THREAD

**Discussion by skidunion** [immich-app/immich#3489] [Feature Request]
### The feature

1. Highlight faces of people by hovering on them

    When hovering over a person in the "People" section of a photo, highlight their face on the picture, for example, with a rectangle.

2. Display people's names when hovering over them on the picture

    Show a tooltip with the person's name when hovering over them. Clicking the tooltip can open all the photos with them.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by fabiannagel** [comment]
+1, also just thought of this. May have some usability and performance hits for large crowd photos, so maybe worth checking out how Google Photos handles this. Would a PR for this be accepted?

**Comment by DrSpaldo** [comment]
Definitely needed. I’ve got a bunch of photos that are being tagged as one person, when there isn’t the same person in a bunch of the photos. So it would be good to see who is actually being tagged as the wrong person. Mostly group photos etc

**Comment by Milkysunshine** [comment]
I agree this would be a fantastic feature. Taking it one step further, clicking on the highlight should allow changing the person identified. 

The same feature should be available for tags!

**Reply by zodcas** [reply]
OP +1
Reply +1

**Comment by fabiannagel** [comment]
@alextran1502 Does the current facial recognition model already provide information about the position of faces?

**Reply by alextran1502** [reply]
Yes, it is already in the database

**Reply by tsemachh** [reply]
@alextran1502 trying now to implement it , can you shed some light or reference about the position and the x1 x2 y1 y2 in the database on which units are they ?

**Comment by akropp** [comment]
+1

**Comment by Th0mC** [comment]
Point 1 is implemented if I understood well.
Point 2 would be the still missing part.

**Comment by cratoo** [comment]
In my duplicate #17986 I suggested to have the option to draw the boxes even without hovering. This would make initial tagging easier as one would see the missing faces at the first glance, as they'd not have a box around them.
